### PR TITLE
chore(edit-content): Import From URL for Image Field

### LINKED ISSUE #30061: Image Field: "Import from URL" functionality
## Parent Issue
#29692

## Task
Implement the "Import from URL" feature for **Image Field**, which opens a popup for users to enter a URL and import a file from that location.

## Proposed Objective
User Experience

## Proposed Priority
Same as Parent Issue

## Acceptance Criteria
1. An "Import from URL" option is available in the File field
2. Clicking the option opens a popup with a URL input field
3. The system attempts to download and import the file from the provided URL
4. Error handling if the URL is invalid or the file cannot be imported
5. The popup remains open and displays an error message if the import fails
6. Successfully imported files are treated as if they were uploaded directly

## Assumptions & Initiation Needs
- Backend API for importing files from URLs is available
- Security measures for URL validation and file scanning are in place

### PR DESCRIPTION
**Pull request by nicobytes**
## Parent Issue

#30061 

## Summary

This pull request includes significant updates to the `DotFormImportUrlComponent` and its associated store, focusing on improving file upload handling and adding comprehensive unit tests. The most important changes include updating the component and store to handle accepted file types, adding abort functionality for uploads, and enhancing test coverage.

### Component and Store Enhancements:

* Updated the `DotFormImportUrlComponent` to handle accepted file types and added an abort controller to manage upload cancellations. 
*  Modified the `FormImportUrlStore` to include accepted file types in its state and added methods to initialize setup and handle file uploads with abort functionality.

### Unit Tests Enhancements:

* Added comprehensive unit tests for the `DotFormImportUrlComponent`, covering initialization, form submission, and upload cancellation scenarios. 

* Added unit tests for the `FormImportUrlStore` to verify state initialization, setup, and file upload handling.

### Minor UI Adjustments:

* Adjusted the error message and button styles to improve the user interface. 

### Test Updates for File Field Component:

* Updated existing tests to use the new `uploadFile` method instead of `uploadDotAsset`.

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

### Video

This PR fixes: #30061

### CODE REVIEW

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-form-import-url/store/form-import-url.store.ts [hunk 1]
```diff
@@ -15,39 +17,64 @@ export interface FormImportUrlState {
...
+    uploadType: 'temp',
+    acceptedFiles: []
 };
 
 export const FormImportUrlStore = signalStore(
     withState(initialState),
     withComputed((state) => ({
         isLoading: computed(() => state.status() === 'uploading'),
-        isDone: computed(() => state.status() === 'done')
+        isDone: computed(() => state.status() === 'done'),
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-form-import-url/store/form-import-url.store.ts:35]: 👊

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-form-import-url/dot-form-import-url.component.html [hunk 2]
```diff
@@ -32,7 +32,7 @@
             type="button"
             aria-label="Cancel button"
             data-testId="cancel-button" />
-        <div>
+        <div [style]="{ width: '6.85rem' }">
```
Comment (hmoreras) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-form-import-url/dot-form-import-url.component.html:35]: add a class instead ?

Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-form-import-url/dot-form-import-url.component.html]: Fixed!

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/services/upload-file/upload-file.service.ts [hunk 3]
```diff
@@ -31,30 +39,79 @@ export class DotFileFieldUploadService {
...
+     * @param file The file to be uploaded.
+     * @param acceptedFiles The accepted mime types.
+     * @returns An observable that resolves to the created contentlet.
+     * If the file type is not in the accepted types, it will throw an error.
+     */
+    uploadDotAssetByFile(file: File, acceptedFiles: string[]): Observable<DotCMSContentlet> {
+        return this.uploadDotAsset(file).pipe(
+            tap((file) => {
+                if (!checkMimeType(file, acceptedFiles)) {
+                    throw new Error('Invalid file type');
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/services/upload-file/upload-file.service.ts:92]: translate?

**Review by oidacra — APPROVED** [review]

**Pull request merged**